File: LinearSystem/src/Interpolation.cpp
#include "Interpolation.h"

Interpolation::Interpolation(int n1) : Matrix(n1, 2)
{}

void Interpolation::readPoints(ifstream &fin)
{
    for(int i = 0; i < rows; i++)
    {
        fin >> (*this)(i,0) >> (*this)(i,1);
    }
}

File: LinearSystem in Python/src/Interpolation.py
from Matrix import Matrix


class Interpolation(Matrix):
    def __init__(self, n1=0):
        super().__init__(n1, 2)

    def read_points(self, fin):
        for i in range(self.rows):
            tokens = fin.readline().split()
            self.a[i][0] = float(tokens[0])
            self.a[i][1] = float(tokens[1])

    def solve(self, xp):
        raise NotImplementedError
